.github/workflows/node.js.yml

```diff
@@ -18,20 +16,31 @@ jobs:
       uses: actions/setup-node@v2
       with:
         node-version: 15.x
-    - run: npm install
-    - run: npm run build --if-present
-    - run: npm test
+    - name: Install Yarn
+      run: npm install -g yarn
+    - run: yarn install
+    - run: yarn run build --if-present
+    - run: yarn test --watchAll=false --coverage
+    - uses: actions/upload-artifact@v1
+      with:
+        name: lcov.info
+        path: ./coverage/lcov.info
+    - uses: actions/upload-artifact@v1
+      with:
+        name: HTML Report
+        path: ./coverage/lcov-report
+    - uses: codecov/codecov-action@v1
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
+        file: ./coverage/lcov.info
+
   deploy:
       name: Deploy with metrics
       runs-on: ubuntu-latest
-      #needs: [build]
+      needs: [build]
       steps:
-        - name: Quickmetrics
-          uses: screendriver/quickmetrics-action@v1
-          with:
-            name: fake-deploy-front
-            api-key: ${{ secrets.QUICKMETRICS_API_KEY }}
         - name: Invoke deployment hook
+          if: github.ref == 'refs/heads/main'
```

Comment:
What happen when this condition is false ? Does it fail the job ?

Reply:
It will only skip the deployment hooks. Like that it will be trigger only during a push on main branch